app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -1029,17 +1029,20 @@ class BrowserTabViewModel @Inject constructor(
             else -> {}
         }
 
-        if (newWebNavigationState.progress ?: 0 >= SHOW_CONTENT_MIN_PROGRESS) {
+        if ((newWebNavigationState.progress ?: 0) >= SHOW_CONTENT_MIN_PROGRESS) {
             showWebContent()
         }
         navigationAwareLoginDetector.onEvent(NavigationEvent.WebNavigationEvent(stateChange))
     }
 
+    override fun onPageContentStart(url: String) {
+        showWebContent()
```

Comment:
Isn't this going to be always executed twice? Isn't it enough to call it from `onPageCommitVisible`?

Reply:
I guess you mean removing `navigationStateChanged` from `onPageCommitVisible`? because it can trigger `showWebContent` twice if progress <50. You are right, but it's fine in favor of executing that stateChanged on that callback as well.